src/components/Modals/WorldPermissionsModal/WorldPermissionsModal.tsx

```diff
@@ -0,0 +1,255 @@
+import WorldPermissionsAccess from './ModelTabs/WorldPermissionsAccess/WorldPermissionsAccess'
+
+import './WorldPermissionsModal.css'
+import WorldPermissionsCollaborators from './ModelTabs/WorldPermissionsCollaborators/WorldPermissionsCollaborators'
+
+enum TabsSection {
+  Access = 'Access',
+  Collaborators = 'Collaborators'
+}
+
+const WorldPermissionsModal = (props: Props) => {
+  const {
+    name,
+    metadata,
+    worldPermissions,
+    profiles,
+    isLoading,
+    isLoadingNewUser,
+    onPutWorldPermissionsRequest,
+    onPostWorldPermissionsRequest,
+    onDeleteWorldPermissionsRequest,
+    onGetProfilesRequest,
+    onClose
+  } = props
+  const [newAddress, setNewAddress] = useState('')
+  const [showAddUserForm, setShowAddUserForm] = useState(false)
+  const [errorInvalidAddress, setErrorInvalidAddress] = useState(false)
+  const [isAccessUnrestricted, setAccessUnrestricted] = useState(worldPermissions?.access.type === WorldPermissionType.Unrestricted)
+  const [collaboratorUserList, setCollaboratorUserList] = useState<string[]>([])
+  const [tabSelected, setTabSelected] = useState<TabsSection>(TabsSection.Access)
+
+  const loading = isLoading || !metadata.worldName || !worldPermissions
+
+  useEffect(() => {
+    setAccessUnrestricted(worldPermissions?.access.type === WorldPermissionType.Unrestricted)
+  }, [worldPermissions?.access.type])
+
+  useEffect(() => {
+    if (
+      tabSelected === TabsSection.Access &&
+      worldPermissions?.access.type === WorldPermissionType.AllowList &&
+      worldPermissions?.access.wallets.length > 0
+    ) {
+      const allowedWallets = new Set(worldPermissions.access.wallets)
+      allowedWallets.has(newAddress.toLowerCase()) && setNewAddress('')
+    }
+  }, [worldPermissions?.access, newAddress, tabSelected])
+
+  useEffect(() => {
+    if (tabSelected === TabsSection.Collaborators && collaboratorUserList.length > 0) {
+      const allowedWallets = new Set(collaboratorUserList)
+      allowedWallets.has(newAddress.toLowerCase()) && setNewAddress('')
+    }
+  }, [collaboratorUserList, newAddress, tabSelected])
+
+  useEffect(() => {
+    setCollaboratorUserList(collaboratorUserList => {
+      const newCollaboratorUserList = new Set(collaboratorUserList)
+
+      if (worldPermissions?.deployment.type === WorldPermissionType.AllowList) {
+        worldPermissions.deployment.wallets.forEach(wallet => {
+          !newCollaboratorUserList.has(wallet) && newCollaboratorUserList.add(wallet.toLowerCase())
+        })
+      }
+      if (worldPermissions?.streaming.type === WorldPermissionType.AllowList) {
+        worldPermissions.streaming.wallets.forEach(wallet => {
+          !newCollaboratorUserList.has(wallet) && newCollaboratorUserList.add(wallet.toLowerCase())
+        })
+      }
+      return [...newCollaboratorUserList]
+    })
+  }, [worldPermissions])
+
+  const profilesByWallets = useMemo(() => profiles, [profiles])
```

Comment:
This shouldn't be necessary, would you mind removing it?
```suggestion
  const profilesByWallets = useMemo(() => profiles, [profiles])
```